.github/workflows/ci-go.yml

```diff
@@ -104,3 +104,44 @@ jobs:
 
       - name: Session tests
         run: make bin/ak test-sessions
+
+  publish_docker_image:
+    needs:
```

Comment:
do we need really these here? i'm good either way.

Reply:
It just make sure we don't publish an image if tests aren't passing

It won't start the publish job until all build and test jobs finish

Reply:
The confusing part for me is that this file runs for all PRs as well as merges into main, and this job has a specific filter: `if: github.ref == 'refs/heads/main'`.

It's better to just put it in a separate file, with this trigger:

```
on:
  push:
    branches:
      - main
```

i.e. without pull requests, and even without depending on tests, since our PRs are required to have linear history and all test passing before merging.